src/Microsoft.ML.Transforms/CategoryImputer.cs

```diff
@@ -0,0 +1,1876 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using Microsoft.ML;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Transforms;
+using Microsoft.Win32.SafeHandles;
+
+[assembly: LoadableClass(typeof(CategoryImputerTransformer), null, typeof(SignatureLoadModel),
+    CategoryImputerTransformer.UserName, CategoryImputerTransformer.LoaderSignature)]
+
+[assembly: EntryPointModule(typeof(CategoryImputerEntrypoint))]
+
+namespace Microsoft.ML.Transforms
+{
+    public static class CategoryImputerExtensionClass
+    {
+        public static CategoryImputerTransformerEstimator CatagoryImputerTransformer(this TransformsCatalog catalog, string outputColumnName, string inputColumnName = null)
+            => CategoryImputerTransformerEstimator.Create(CatalogUtils.GetEnvironment(catalog), outputColumnName, inputColumnName);
+
+        public static CategoryImputerTransformerEstimator CatagoryImputerTransformer(this TransformsCatalog catalog, params InputOutputColumnPair[] columns)
+            => CategoryImputerTransformerEstimator.Create(CatalogUtils.GetEnvironment(catalog), columns);
+    }
+
+    public sealed class CategoryImputerTransformerEstimator : IEstimator<CategoryImputerTransformer>
+    {
+        private Options _options;
+
+        private readonly IHost _host;
+
+        #region Options
+        internal sealed class Column : OneToOneColumn
+        {
+            internal static Column Parse(string str)
+            {
+                Contracts.AssertNonEmpty(str);
+
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            internal bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        internal sealed class Options: TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition (optional form: name:src)",
+                Name = "Column", ShortName = "col", SortOrder = 1)]
+            public Column[] Columns;
+        }
+        #endregion
+
+        // Using this to confirm DLL exists. If does it will just return false since no parameters are being passed.
```

Comment:
Why do we need that? Isn't the DLL is part of the package?

Reply:
It will be a dependency we get through nuget, but until we get the nuget up and running this is in there just as a check. Once the nuget is ready we will remove this check.